packages/inferno-router/src/HashRouter.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * @module Inferno-Router
+ */ /** TypeDoc Comment */
+
+import { createVNode, VNode, Component } from "inferno";
+import VNodeFlags from "inferno-vnode-flags";
+import createHistory from "history/createHashHistory";
+import Router from "./Router";
+import { warning } from "./utils";
+
+export interface IHashRouterProps {
+  basename: string;
+  getUserConfirmation: () => {};
+  hashType: string;
+  children: Component<any, any>[];
+}
+
+export default class HashRouter extends Component<IHashRouterProps, any> {
+  public history;
+
+  constructor(props?: any, context?: any) {
+    super(props, context);
+    this.history = createHistory(props);
+  }
+
+  public componentWillMount() {
+    warning(
```

Comment:
this componentWillMount function could be behind process.env.NODE_ENV !== 'production'  to reduce production size of library